### DISCUSSION TITLE
Challenges during initial setup USB-C to MacBook Port Configuration during initial Machine Configuration

### DISCUSSION THREAD

**Discussion by RCAnthonyClark** [artisan-roaster-scope/artisan#1143] [General]
Problems Artisan software MACOS Hottop kn-8828b-2k+ 

Using the provided USB 2 cable from Hottop, I repeatedly encountered problems with my initial configuration of the comm port.  After letting the roaster start up in AUTO mode. Eventually, (but not at first), if you are patient, the "Port Configuration" window option to select "FT30 Basic UART" will eventually appear.  This is the setting that must be selected, all other setting will not work (e.g. /dev/cu.BLTH and /dev/cu.usbserial-DA01POGR)

**Comment by poundy** [comment]
so do you have a problem or was this more just a post to show others? 
Even if only for someone's info it would be beneficial for you to document the age of your hardware and the MacOS version you're running.

**Reply by RCAnthonyClark** [reply]
I posted this to help others; after reviewing my post, I could have spent a little more time and been a lot clearer.

Hottop Roaster purchased in 4/2022, running current artisan software 2.2.2, on MacBook Pro 2019 running current OS 13.3

**Comment by RCAnthonyClark** [comment]
Problem solved; happy to augment and add to what is posted. Thanks for the feedback!-RaySent from my iPhoneOn Apr 4, 2023, at 5:02 PM, poundy ***@***.***> wrote:﻿
so do you have a problem or was this more just a post to show others?
Even if only for someone's info it would be beneficial for you to document the age of your hardware and the MacOS version you're running.

**Comment by RCAnthonyClark** [comment]
From the information that I was able to locate, with the latest version of the macOS operating system, drivers are no longer needed. I’m not sure how far back that goes but it definitely is the case for the present.

The secret sauce for getting Artisian running on your computer, and recognizing your roaster device is giving the comm port time to analyze the comm connection.   This is probably a function of the MacOS; and this can only happen with roaster on, laptop on, comm cable connected, start Artisan software and config your machine.

It didn’t happen immediately for me but after a few iterations of ending the artisan software, restarting artisan and configuring a new machine, and ultimately the comm port UART option (or another option may show up for your roaster) will appear while configuring the comm port where it never had before.  The UART option does not appear right away, and once it does, stop the (test) roast on artisan and save the profile as your base UART step one.

Next time you start your roast, load this profile upon starting Artisan and it will connect every time immediately.  Obviously there are other parameters to set depending on your roaster, once you configure these and save them, you will have your base configuration to start your roast with all the machine parameters except for the particulars about your roast and modifications you may desire to change or control.

But ‘till you have comms, nothing else matters.

Happy roasting!

Ray

From: Dave Lehman ***@***.***>
Reply-To: artisan-roaster-scope/artisan ***@***.***>
Date: Monday, April 17, 2023 at 9:37 AM
To: artisan-roaster-scope/artisan ***@***.***>
Cc: Raymond Clark ***@***.***>, Author ***@***.***>
Subject: Re: [artisan-roaster-scope/artisan] Challenges during initial setup USB-C to MacBook Port Configuration during initial Machine Configuration (Discussion #1143)


Thanks for posting. I have an M1 MacBook Pro running Ventura (13.3) w/ Artisan 2.8.2 i'm trying to connect to a Toper 1kg roaster. Can't get it connected properly to Artisan, or displaying any readings.

Did you have to install drivers on the Mac to get it to recognize/show the COM port? Or did it just show up after you plugged it in to laptop? If I plug USB cable to laptop, and run an ioreg... command, I can see PhidgetTemperatureSensor device (that disappears as soon as I unplug the cable), but it does not show up as a /dev/cu.* or /dev/cu.usb* device. So not sure if i'm on the right path or not.
